docs/understand/architecture.md

```diff
@@ -57,6 +72,35 @@ The credential scheme, along with the request signing logic are compatible with
 
 Currently, the auth service manages its own database of users and credentials and does not use IAM in any way. 
 
+### Hooks Engine
+
+The hooks engine enables CI/CD for data by triggering user defined [Actions](../setup/hooks.md) that will run during commit/merge. 
+
 ### Frontend UI
 
 The UI layer is a simple browser-based client that uses the OpenAPI server. It allows management, exploration and data access to repositories, branches, commits and objects in the system.
+
+## lakeFS Clients
+
+There are several ways of interacting with lakeFS. Many data applications have different use-cases for lakeFS and 
+we try to support as many as possible.
```

Comment:
```suggestion
## Applications 

As a rule of thumb, lakeFS supports any s3-compatible application. This means that many common data applications work with lakeFS out of the box. 

## lakeFS Clients

Some data applications require deeper integrations with lakeFS to support different use-cases or enhanced functionality which is provided by lakeFS clients.
```

Feel free to change the wording here but I tried to add details I think worth including.

Reply:
Done.